src/test/java/uk/gov/companieshouse/officerfiling/api/service/TransactionServiceImplTest.java

```diff
@@ -22,47 +28,82 @@
     @Mock
+    private PrivateTransactionResourceHandler privateTransactionResourceHandler;
+    @Mock
     private ApiResponse<Transaction> apiResponse;
+    @Mock
+    private ApiResponse<Void> apiResponseVoid;
+    @Mock
+    private PrivateTransactionPatch privateTransactionPatch;
 
+    private Transaction testTransaction;
     private TransactionServiceImpl testService;
 
     @BeforeEach
     void setUp() {
         testService = new TransactionServiceImpl(apiClientService);
+        testTransaction = testTransaction(TRANS_ID);
     }
 
     @Test
-    public void getTransactionWhenFound() throws IOException, URIValidationException {
-        var transactionId = "12345";
-
-        when(apiResponse.getData()).thenReturn(testTransaction(transactionId));
+    void getTransactionWhenFound() throws IOException, URIValidationException {
+        when(apiResponse.getData()).thenReturn(testTransaction(TRANS_ID));
         when(transactionsGet.execute()).thenReturn(apiResponse);
-        when(transactionsResourceHandler.get("/transactions/" + transactionId)).thenReturn(transactionsGet);
+        when(transactionsResourceHandler.get("/transactions/" + TRANS_ID)).thenReturn(transactionsGet);
         when(apiClient.transactions()).thenReturn(transactionsResourceHandler);
-        when(apiClientService.getOauthAuthenticatedClient(PASS_THOROUGH)).thenReturn(apiClient);
+        when(apiClientService.getOauthAuthenticatedClient(PASSTHROUGH_HEADER)).thenReturn(apiClient);
+
+        var transaction = testService.getTransaction(TRANS_ID, PASSTHROUGH_HEADER);
+
+        assertThat(transaction, samePropertyValuesAs(testTransaction(TRANS_ID)));
+    }
+
+    @Test
+    void getTransactionWhenNotFound() throws IOException, URIValidationException {
+        when(apiClientService.getOauthAuthenticatedClient(PASSTHROUGH_HEADER)).thenThrow(IOException.class);
+
+        assertThrows(TransactionServiceException.class, () -> testService.getTransaction(TRANS_ID, PASSTHROUGH_HEADER));
+    }
+
+    @Test
+    void updateTransactionWhenResponse204() throws IOException, URIValidationException {
+        when(privateTransactionPatch.execute()).thenReturn(apiResponseVoid);
+        when(privateTransactionResourceHandler.patch("/private/transactions/12345", testTransaction)).thenReturn(
+                privateTransactionPatch);
+        when(internalApiClient.privateTransaction()).thenReturn(privateTransactionResourceHandler);
+        when(apiClientService.getInternalOauthAuthenticatedClient(PASSTHROUGH_HEADER)).thenReturn(internalApiClient);
 
-        var transaction = testService.getTransaction(transactionId, "passThorough");
+        when(apiResponseVoid.getStatusCode()).thenReturn(HttpStatusCodes.STATUS_CODE_NO_CONTENT);
+        testService.updateTransaction(testTransaction, PASSTHROUGH_HEADER);
 
-        assertThat(transaction, samePropertyValuesAs(testTransaction(transactionId)));
+        verify(apiResponseVoid, times(1)).getStatusCode();
```

Comment:
ℹ️ `times(1)` used by default, so it's optional to use as second arg.

Reply:
amended as described